shell/src/main/java/alluxio/cli/fs/command/LsCommand.java

```diff
@@ -192,47 +227,47 @@ private void ls(AlluxioURI path, boolean recursive, boolean forceLoadMetadata, b
     if (forceLoadMetadata) {
       options.setLoadMetadataType(LoadMetadataType.Always);
     }
-    List<URIStatus> statuses = listStatusSortedByIncreasingCreationTime(path, options);
-    for (URIStatus status : statuses) {
+    List<URIStatus> statuses = mFileSystem.listStatus(path, options);
+    List<URIStatus> sorted = sortByFieldAndOrder(statuses, sortField, reverse);
+    for (URIStatus status : sorted) {
       if (!pinnedOnly || status.isPinned()) {
         printLsString(status, hSize);
       }
       if (recursive && status.isFolder()) {
         ls(new AlluxioURI(path.getScheme(), path.getAuthority(), status.getPath()), true,
-            forceLoadMetadata, false, hSize, pinnedOnly);
+            forceLoadMetadata, false, hSize, pinnedOnly, sortField, reverse);
       }
     }
   }
 
-  private List<URIStatus> listStatusSortedByIncreasingCreationTime(AlluxioURI path,
-      ListStatusOptions options) throws AlluxioException, IOException {
-    List<URIStatus> statuses = mFileSystem.listStatus(path, options);
-    Collections.sort(statuses, new Comparator<URIStatus>() {
-      @Override
-      public int compare(URIStatus status1, URIStatus status2) {
-        long t1 = status1.getCreationTimeMs();
-        long t2 = status2.getCreationTimeMs();
-        if (t1 < t2) {
-          return -1;
-        }
-        if (t1 == t2) {
-          return 0;
-        }
-        return 1;
-      }
-    });
-    return statuses;
+  private List<URIStatus> sortByFieldAndOrder(
+          List<URIStatus> statuses, String sortField, boolean reverse) throws IOException {
+    Optional<Comparator<URIStatus>> sortToUse = Optional.ofNullable(
```

Comment:
is this the same as returning the `nullable` and then checking `!= null` instead of `isPresent()`?

Reply:
Yes. In java 8, I believe this is the preferred way.